dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/oracle/StatsTaskListGenerator.java

```diff
@@ -16,17 +16,64 @@
  */
 package com.google.edwmigration.dumper.application.dumper.connector.oracle;
 
+import com.google.auto.value.AutoValue;
 import com.google.common.collect.ImmutableList;
+import com.google.common.io.Resources;
 import com.google.edwmigration.dumper.application.dumper.ConnectorArguments;
+import com.google.edwmigration.dumper.application.dumper.task.JdbcSelectTask;
 import com.google.edwmigration.dumper.application.dumper.task.Task;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
 import javax.annotation.Nonnull;
 import javax.annotation.ParametersAreNonnullByDefault;
 
 @ParametersAreNonnullByDefault
 class StatsTaskListGenerator {
 
+  private static final ImmutableList<StatsQuery> QUERIES =
+      ImmutableList.of(
+          // TODO: add entries for other SQLs to this list
+          StatsQuery.create("hist-cmd-types", Tool.STATSPACK));
+
   @Nonnull
-  ImmutableList<Task<?>> createTasks(ConnectorArguments arguments) {
-    throw new UnsupportedOperationException();
+  ImmutableList<Task<?>> createTasks(ConnectorArguments arguments) throws IOException {
+    ImmutableList.Builder<Task<?>> builder = ImmutableList.<Task<?>>builder();
+
+    for (StatsQuery item : QUERIES) {
+      builder.add(item.toTask());
+    }
+    return builder.build();
+  }
+
+  enum Tool {
+    AWR("awr"),
+    STATSPACK("statspack");
+
+    final String value;
+
+    Tool(String value) {
+      this.value = value;
+    }
+  }
+
+  @AutoValue
+  abstract static class StatsQuery {
+
+    abstract String name();
+
+    abstract Tool tool();
+
+    static StatsQuery create(String name, Tool tool) {
+      return new AutoValue_StatsTaskListGenerator_StatsQuery(name, tool);
+    }
+
+    @Nonnull
+    Task<?> toTask() throws IOException {
+      String path = String.format("oracle-stats/%s/%s", tool().value, name());
```

Comment:
There is also a third group of SQL files - metadata that are neither AWS or STATSPACK data. Those files should be stored in the ZIP root directory. Feel free to address it later.

Reply:
Acknowledged, it shouldn't be a problem to add a value later, since the Tool enum works just as a wrapper for the strings.